contracts/integrations/test/exchange/fill_dydx_order_test.ts

```diff
@@ -0,0 +1,133 @@
+import { DydxBridgeActionType, dydxBridgeDataEncoder, TestDydxBridgeContract } from '@0x/contracts-asset-proxy';
+import { DummyERC20TokenContract } from '@0x/contracts-erc20';
+import { blockchainTests, constants, describe, expect, toBaseUnitAmount } from '@0x/contracts-test-utils';
+import { BigNumber } from '@0x/utils';
+import { DecodedLogArgs, LogEntry, LogWithDecodedArgs } from 'ethereum-types';
+import * as _ from 'lodash';
+
+import { deployDydxBridgeAsync } from '../bridges/deploy_dydx_bridge';
+import { Actor } from '../framework/actors/base';
+import { Maker } from '../framework/actors/maker';
+import { Taker } from '../framework/actors/taker';
+import { DeploymentManager } from '../framework/deployment_manager';
+
+blockchainTests.resets('Exchange fills dydx orders', env => {
+    let testContract: TestDydxBridgeContract;
+    let makerToken: DummyERC20TokenContract;
+    let takerToken: DummyERC20TokenContract;
+    const marketId = new BigNumber(3);
+    const makerAssetAmount = toBaseUnitAmount(6);
+    const takerAssetAmount = toBaseUnitAmount(1);
+    let maker: Maker;
+    let taker: Taker;
+    const defaultDepositAction = {
+        actionType: DydxBridgeActionType.Deposit as number,
+        accountId: constants.ZERO_AMOUNT,
+        marketId,
+        // The bridge is passed the `makerAssetFillAmount` and we
+        // want to compute the input `takerAssetFillAmount`
+        //   => multiply by `takerAssetAmount` / `makerAssetAmount`.
+        conversionRateNumerator: takerAssetAmount,
+        conversionRateDenominator: makerAssetAmount,
+    };
+    const defaultWithdrawAction = {
+        actionType: DydxBridgeActionType.Withdraw as number,
+        accountId: constants.ZERO_AMOUNT,
+        marketId,
+        conversionRateNumerator: constants.ZERO_AMOUNT,
+        conversionRateDenominator: constants.ZERO_AMOUNT,
+    };
+    const bridgeData = {
+        accountNumbers: [new BigNumber(0)],
+        actions: [defaultDepositAction, defaultWithdrawAction],
+    };
+
+    before(async () => {
+        // Deploy contracts
+        const deployment = await DeploymentManager.deployAsync(env, {
+            numErc20TokensToDeploy: 2,
+        });
+        testContract = await deployDydxBridgeAsync(deployment, env);
+        const encodedBridgeData = dydxBridgeDataEncoder.encode({ bridgeData });
+        const bridgeProxyAssetData = deployment.assetDataEncoder
+            .ERC20Bridge(testContract.address, testContract.address, encodedBridgeData)
+            .getABIEncodedTransactionData();
+        [makerToken, takerToken] = deployment.tokens.erc20;
+
+        // Configure Maker & Taker.
+        const orderConfig = {
```

Comment:
We should consider adding some tests for dydx bridge assetData as the `takerAssetData`, `makerFeeAssetData`, `takerFeeAssetData` to be safe

Reply:
Yeah, this could be good to demonstrate how the bridge would be used in other ways.